store/sqlstore/retention_policy_store.go

```diff
@@ -519,62 +619,103 @@ func (s *SqlRetentionPolicyStore) AddTeams(policyId string, teamIds []string) er
 	if err := s.checkTeamsExist(teamIds); err != nil {
 		return err
 	}
-	builder := s.getQueryBuilder().
+	query := s.getQueryBuilder().
 		Insert("RetentionPoliciesTeams").
 		Columns("PolicyId", "TeamId")
 	for _, teamId := range teamIds {
-		builder = builder.Values(policyId, teamId)
+		query = query.Values(policyId, teamId)
+	}
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return errors.Wrap(err, "retention_policies_teams_tosql")
 	}
-	_, err := builder.RunWith(s.GetMaster()).Exec()
-	return err
+
+	if _, err := s.GetMasterX().Exec(queryString, args...); err != nil {
+		return errors.Wrap(err, "failed to insert retention policies teams")
+	}
+
+	return nil
 }
 
 func (s *SqlRetentionPolicyStore) RemoveTeams(policyId string, teamIds []string) error {
 	if len(teamIds) == 0 {
 		return nil
 	}
-	builder := s.getQueryBuilder().
+	query := s.getQueryBuilder().
 		Delete("RetentionPoliciesTeams").
 		Where(sq.And{
 			sq.Eq{"PolicyId": policyId},
 			sq.Eq{"TeamId": teamIds},
 		})
-	_, err := builder.RunWith(s.GetMaster()).Exec()
-	return err
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return errors.Wrap(err, "retention_policies_teams_tosql")
+	}
+
+	if _, err := s.GetMasterX().Exec(queryString, args...); err != nil {
+		return errors.Wrapf(err, "unable to permanent delete retention policies teams with policyid=%s", policyId)
+	}
+
+	return nil
 }
 
 // DeleteOrphanedRows removes entries from RetentionPoliciesChannels and RetentionPoliciesTeams
 // where a channel or team no longer exists.
 func (s *SqlRetentionPolicyStore) DeleteOrphanedRows(limit int) (deleted int64, err error) {
 	// We need the extra level of nesting to deal with MySQL's locking
-	const rpcDeleteQuery = `
-	DELETE FROM RetentionPoliciesChannels WHERE ChannelId IN (
-		SELECT * FROM (
-			SELECT ChannelId FROM RetentionPoliciesChannels
-			LEFT JOIN Channels ON RetentionPoliciesChannels.ChannelId = Channels.Id
-			WHERE Channels.Id IS NULL
-			LIMIT :Limit
-		) AS A
-	)`
-	const rptDeleteQuery = `
-	DELETE FROM RetentionPoliciesTeams WHERE TeamId IN (
-		SELECT * FROM (
-			SELECT TeamId FROM RetentionPoliciesTeams
-			LEFT JOIN Teams ON RetentionPoliciesTeams.TeamId = Teams.Id
-			WHERE Teams.Id IS NULL
-			LIMIT :Limit
-		) AS A
-	)`
-	props := map[string]interface{}{"Limit": limit}
-	result, err := s.GetMaster().Exec(rpcDeleteQuery, props)
+	rpcSubQuery := s.getQueryBuilder().
+		Select("ChannelId").
+		From("RetentionPoliciesChannels").
+		LeftJoin("Channels ON RetentionPoliciesChannels.ChannelId = Channels.Id").
+		Where("Channels.Id IS NULL").
+		Limit(uint64(limit))
+
+	rpcSubQueryString, rpcArgs, err := rpcSubQuery.ToSql()
+	if err != nil {
+		return int64(0), errors.Wrap(err, "retention_policies_channels_tosql")
+	}
+
+	rpcDeleteQuery, _, err := s.getQueryBuilder().
+		Delete("RetentionPoliciesChannels").
+		Where(`ChannelId IN (SELECT * FROM (` + rpcSubQueryString + `) AS A)`).
```

Comment:
Can we not reuse the subquery directly here? I thought it only conflicted if there were params in the parent query also. Take a look here: https://github.com/Masterminds/squirrel/issues/258#issuecomment-673315028